03_Modules/OcppRouter/src/module/router.ts

```diff
@@ -252,6 +262,71 @@ export class MessageRouterImpl extends AbstractMessageRouter implements IMessage
      * Private Methods 
      */
 
+
+    /**
+     * Loads all subscriptions for a given connection into memory
+     *
+     * @param {string} connectionIdentifier - the identifier of the connection
+     * @return {Promise<void>} a promise that resolves once all subscriptions are loaded
+     */
+    private async _loadSubscriptionsForConnection(connectionIdentifier: string) {
+        this._onConnectionCallbacks.set(connectionIdentifier, []);
+        this._onCloseCallbacks.set(connectionIdentifier, []);
+        this._onMessageCallbacks.set(connectionIdentifier, []);
+        this._sentMessageCallbacks.set(connectionIdentifier, []);
+        const subscriptions = await this.subscriptionRepository.readAllByStationId(connectionIdentifier);
+        for (const subscription of subscriptions) {
+            if (subscription.onConnect) {
+                this._onConnectionCallbacks.get(connectionIdentifier)!.push((info?: Map<string, string>) => this._subscriptionCallback({ stationId: connectionIdentifier, event: 'connected', info: info }, subscription.url));
+                this._logger.debug(`Added onConnect callback to ${subscription.url} for station ${connectionIdentifier}`);
+            }
+            if (subscription.onClose) {
+                this._onCloseCallbacks.get(connectionIdentifier)!.push((info?: Map<string, string>) => this._subscriptionCallback({ stationId: connectionIdentifier, event: 'closed', info: info }, subscription.url));
+                this._logger.debug(`Added onClose callback to ${subscription.url} for station ${connectionIdentifier}`);
+            }
+            if (subscription.onMessage) {
+                this._onMessageCallbacks.get(connectionIdentifier)!.push(async (message: string, info?: Map<string, string>) => {
+                    if (!subscription.messageRegexFilter || new RegExp(subscription.messageRegexFilter).test(message)) {
+                        return this._subscriptionCallback({ stationId: connectionIdentifier, event: 'message', origin: MessageOrigin.ChargingStation, message: message, info: info }, subscription.url)
```

Comment:
Do we need to differentiate the event 'onMessage' and 'sentMessage'?

Reply:
Good question. It made sense to me to keep them separate since they occur at different places in the code. In terms of use case, I can see situations where you may only want one or the other, such as if you just want to monitor StatusNotificationRequest and the response (which is empty) is pointless.